vault/resource_terraform_cloud_secret_backend.go

```diff
@@ -0,0 +1,246 @@
+
+	configPath := terraformCloudSecretBackendConfigPath(path)
+
+	info := &api.MountInput{
+		Type:        "terraform",
+		Description: d.Get("description").(string),
+		Config: api.MountConfigInput{
+			DefaultLeaseTTL: fmt.Sprintf("%ds", d.Get("default_lease_ttl_seconds")),
+			MaxLeaseTTL:     fmt.Sprintf("%ds", d.Get("max_lease_ttl_seconds")),
+		},
+	}
+
+	d.Partial(true)
+	log.Printf("[DEBUG] Mounting Terraform Cloud backend at %q", path)
+
+	if err := client.Sys().Mount(path, info); err != nil {
+		return fmt.Errorf("Error mounting to %q: %s", path, err)
+	}
+
+	log.Printf("[DEBUG] Mounted Terraform Cloud backend at %q", path)
+	d.SetId(path)
+
+	d.SetPartial("path")
+	d.SetPartial("description")
+	d.SetPartial("default_lease_ttl_seconds")
+	d.SetPartial("max_lease_ttl_seconds")
+
+	log.Printf("[DEBUG] Writing Terraform Cloud configuration to %q", configPath)
+	data := map[string]interface{}{
+		"address":   address,
+		"token":     token,
+		"base_path": basePath,
+	}
+	if _, err := client.Logical().Write(configPath, data); err != nil {
+		return fmt.Errorf("Error writing Terraform Cloud configuration for %q: %s", path, err)
+	}
+	log.Printf("[DEBUG] Wrote Terraform Cloud configuration to %q", configPath)
+	d.SetPartial("address")
+	d.SetPartial("token")
+	d.SetPartial("base_path")
+	d.Partial(false)
+
+	return nil
+}
+
+func terraformCloudSecretBackendRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	path := d.Id()
+	configPath := terraformCloudSecretBackendConfigPath(path)
+
+	log.Printf("[DEBUG] Reading Terraform Cloud backend mount %q from Vault", path)
+
+	mounts, err := client.Sys().ListMounts()
+	if err != nil {
+		return fmt.Errorf("Error reading mount %q: %s", path, err)
+	}
+
+	// path can have a trailing slash, but doesn't need to have one
+	// this standardises on having a trailing slash, which is how the
+	// API always responds.
+	mount, ok := mounts[strings.Trim(path, "/")+"/"]
+	if !ok {
+		log.Printf("[WARN] Mount %q not found, removing from state.", path)
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("path", path)
+	d.Set("description", mount.Description)
+	d.Set("default_lease_ttl_seconds", mount.Config.DefaultLeaseTTL)
+	d.Set("max_lease_ttl_seconds", mount.Config.MaxLeaseTTL)
+
+	log.Printf("[DEBUG] Reading %s from Vault", configPath)
+	secret, err := client.Logical().Read(configPath)
+	if err != nil {
+		return fmt.Errorf("error reading from Vault: %s", err)
+	}
+
+	log.Printf("[DEBUG] secret: %#v", secret)
+
+	// token, sadly, we can't read out
+	// the API doesn't support it
+	// So... if it drifts, it drift.
+	d.Set("address", secret.Data["address"].(string))
+	d.Set("base_path", secret.Data["base_path"].(string))
+
+	return nil
+}
+
+func terraformCloudSecretBackendUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	path := d.Id()
+	configPath := terraformCloudSecretBackendConfigPath(path)
+
+	d.Partial(true)
```

Comment:
`Partial` is also deprecated and not necessary:

- https://github.com/hashicorp/terraform-plugin-sdk/blob/9d2850173e33bf6e4b4fa0fd0819f602b45e23c8/helper/schema/resource_data.go#L163

Reply:
Removal of partial states will commence post-haste

Reply:
Partial states removed.